### PR TITLE
Handle re-exported modules when constructing ExportsMap

### PR DESCRIPTION
**Pull request by jhrcek**
Attempt at fixing https://github.com/haskell/haskell-language-server/issues/2451 based on @pepeiborra 's suggestion: https://github.com/haskell/haskell-language-server/issues/2451#issuecomment-991637640

This fixes the issue for me with GHC 8.10.7, let's see how it fares in CI..

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2468"></a>

### CODE REVIEW

File: ghcide/src/Development/IDE/Types/HscEnvEq.hs [hunk 1]
```diff
@@ -76,22 +76,25 @@ newHscEnvEqWithImportPaths envImportPaths hscEnv deps = do
...
             depends = explicitUnits pkgst
-            targets =
-                [ (pkg, mn)
+            modules =
+                [ m
                 | d        <- depends
                 , Just pkg <- [lookupPackageConfig d hscEnv]
-                , (mn, _)  <- unitExposedModules pkg
+                , (modName, maybeOtherPkgMod) <- unitExposedModules pkg
+                , m <- pure $ case maybeOtherPkgMod of
```
Comment (pepeiborra) [inline ghcide/src/Development/IDE/Types/HscEnvEq.hs:84]: Why not use `let` here? Something like
```suggestion
                , let m =  case maybeOtherPkgMod of
```

Comment (jhrcek) [reply ghcide/src/Development/IDE/Types/HscEnvEq.hs]: I'm not used to using list comprehensions so much, so it didn't occur to me

**Review by pepeiborra — APPROVED** [review]
Thanks for reporting, diagnosing and fixing this!

**Comment by jhrcek** [comment]
I don't understand what the failure in the windows job it about.
What do you usually do / what's the right thing to do next?

**Pull request merged**